src/dags/bouwstroompunten.py

```diff
@@ -89,89 +91,98 @@ def get_data():
     mkdir = BashOperator(task_id="mkdir", bash_command=f"mkdir -p {tmp_dir}")
 
     # 3. Download data
-    download_data = PythonOperator(task_id="download_data", python_callable=get_data)
+    download_data = HttpFetchOperator(
+        task_id="download",
+        endpoint=data_endpoint,
+        http_conn_id="BOUWSTROOMPUNTEN_BASE_URL",
+        tmp_file=data_file,
+        headers={
+            "content-type": "application/json",
+            "Authorization": f"Bearer {get_token()}",
+        },
+    )
 
-    # 4. Create SQL
+    # 4. Import data
     # ogr2ogr demands the PK is of type intgger. In this case the source ID is of type varchar.
     # So FID=ID cannot be used.
-    create_SQL = BashOperator(
-        task_id="create_SQL_based_on_geojson",
-        bash_command="ogr2ogr -f 'PGDump' "
-        "-t_srs EPSG:28992 "
-        f"-nln {dag_id} "
-        f"{tmp_dir}/{dag_id}.sql {data_file} "
-        "-lco GEOMETRY_NAME=geometry ",
-    )
-
-    # 5. Create TABLE
-    create_table = BashOperator(
-        task_id="create_table",
-        bash_command=f"psql {pg_params()} < {tmp_dir}/{dag_id}.sql",
+    import_data = Ogr2OgrOperator(
+        task_id="import_data",
+        target_table_name=f"{dag_id}_{dag_id}_new",
+        input_file=data_file,
+        s_srs=None,
+        fid="ogc_fid",
+        auto_detect_type="YES",
+        mode="PostgreSQL",
+        db_conn=db_conn,
     )
 
-    # 6. Drop Exisiting TABLE
+    # 5. Drop Exisiting TABLE
     drop_tables = PostgresOperator(
         task_id="drop_existing_table",
         sql=[
             f"DROP TABLE IF EXISTS {dag_id}_{dag_id} CASCADE",
```

Comment:
I know this is existing code, but as you are touching the DAG anyway ...

We should never be using regular string formatting for creating SQL strings. Fortunately we are not processing user input here. Hence the potential for SQL injection is extremely low. But code gets copied around or mimicked hence it should be done properly.

The PostgresqlOperator has its own SQL string formatting capabilities. So it is best to use those:

```python
    drop_tables = PostgresOperator(
        task_id="drop_existing_table",
        sql=[
            "DROP TABLE IF EXISTS {{ params.table_id }} CASCADE",
        ],
        params={"table_id": f"{dag_id}_{dag_id}"} 
    )
```

Reply:
Ok check!